dbcon/mysql/ha_mcs_dml.cpp

```diff
@@ -758,7 +758,7 @@ int ha_mcs_impl_write_batch_row_(const uchar* buf, TABLE* table, cal_impl_if::ca
     rc = fprintf(ci.filePtr, "\n"); //@bug 6077 check whether thhe pipe is still open
 
     if ( rc < 0)
-        rc = -1;
+        rc = errno;
```

Comment:
Really? IIRC we return this value up the call stack back to MDB that knows nothing regarding errno values after fprintf errors.

Reply:
I can't recall why I had to do this. I do know it became an issue.

Reply:
We can't let this pass with this patch. It looks unrelated.